TeamSnapSDK/SDK/DataLayer/TSDKDuplicateCompletionBlockStore.m

```diff
@@ -51,30 +61,42 @@ - (void)addCompletionBlock:(TSDKJSONCompletionBlock)completionBlock forRequest:(
         return;
     }
     
-    if([[self.completionBlocksKeyedByRequest objectForKey:[TSDKDuplicateCompletionBlockStore identifierFromRequest:request]] isKindOfClass:[NSMutableSet class]]) {
-        NSMutableSet* set = [self.completionBlocksKeyedByRequest objectForKey:[TSDKDuplicateCompletionBlockStore identifierFromRequest:request]];
-        [set addObject:completionBlock];
-        
-    } else {
-        NSMutableSet *set = [NSMutableSet setWithObject:completionBlock];
-        [self.completionBlocksKeyedByRequest setObject:set forKey:[TSDKDuplicateCompletionBlockStore identifierFromRequest:request]];
-    }
+    NSString *requestKey = [TSDKDuplicateCompletionBlockStore identifierFromRequest:request];
+    
+    dispatch_sync(self.accessQueue, ^{
+        NSMutableSet *existingCompletions = [self.completionBlocksKeyedByRequest objectForKey:requestKey];
+        if(existingCompletions != nil) {
+            [existingCompletions addObject:completionBlock];
```

Comment:
Good catch. Added `completionBlock == nil` to the fallthrough condition at the start of this method.